providers/implementations/kem/mlx_kem.c

```diff
@@ -287,7 +314,9 @@ static int mlx_kem_decapsulate(void *vctx, uint8_t *shsec, size_t *slen,
     decap_slen = ML_KEM_SHARED_SECRET_BYTES;
     cbuf = ctext + ml_kem_slot * key->xinfo->pubkey_bytes;
     sbuf = shsec + ml_kem_slot * key->xinfo->shsec_bytes;
-    ctx = EVP_PKEY_CTX_new_from_pkey(key->libctx, key->mkey, key->propq);
+    adjusted_propq = get_adjusted_propq(key->propq);
```

Comment:
Same issue here - check NULL and return an error

Reply:
Fixed, thanks